### PR TITLE
Seatbelt: fall back to policy-allowed cwd to avoid getcwd warnings

### PR DESCRIPTION
**Pull request by richiemsft**
## 📖 Description
When an explicit process.cwd (or the inherited host cwd) is not readable under the deny-by-default Seatbelt profile, the child shell's startup getcwd() walk fails and leaks noisy "cannot access parent directories" warnings from bash's shell-init / job-working-directory onto stderr.

resolve_working_directory now honors an explicit working directory only when it is readable under the filesystem policy (within a readwrite/readonly path and not within a denied path, matched component-wise). Otherwise it launches from a policy-allowed directory (first readwrite, else first readonly, else /) and logs an informational diagnostic to the mxc log. The command still runs unchanged; only the launch directory changes, never the granted filesystem access.

Adds unit tests for the allow/deny/fallback/tilde/component-boundary cases and an end-to-end characterization test asserting an out-of-policy cwd produces no getcwd noise while the command still succeeds. Updates the Seatbelt backend doc.

## 🔗 References
Resolves the Mac issue from #611 

## 🔍 Validation
Validated locally on a Mac. All green

## ✅ Checklist

- [x] Signed the [Contributor License Agreement](https://cla.opensource.microsoft.com)
- [x] Linked to an issue
- [x] Updated documentation (if applicable)
- [ ] Updated [Copilot instructions](.github/copilot-instructions.md) (if build, architecture, or conventions changed)
- [ ] If this PR changes `Cargo.lock`, the `dependency-feed-check` check passes (see [docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md))

## 📋 Issue Type

- [x] Bug fix
- [ ] Feature
- [ ] Task

---

GitHub Actions runs the PR validation build automatically. The ADO pipeline
(`MXC-PR-Build`) is the Azure version of the PR pipeline, kept in parity with the GitHub
Actions build; it runs on merge to `main`, and Microsoft reviewers with write access can trigger it
on a PR with `/azp run`. See [docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md).

If the `dependency-feed-check` check fails on a new dependency, the crate must be added to
the feed before the PR can pass. See [docs/pull-requests.md](https://github.com/microsoft/mxc/blob/main/docs/pull-requests.md)
for the steps.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/mxc/pull/688)

### CODE REVIEW

File: src/backends/seatbelt/common/src/seatbelt_runner.rs [hunk 1]
```diff
@@ -708,6 +734,103 @@ fn resolve_working_directory(request: &ExecutionRequest) -> String {
...
+/// Whether `dir` is readable under the sandbox's filesystem policy — i.e. a
+/// process launched there can `getcwd()` without the profile denying the walk.
+///
+/// A directory is considered readable when it is within (a subpath of, or
+/// equal to) some `readwritePaths` / `readonlyPaths` entry and is not within
+/// any `deniedPaths` entry (deny overrides allow, matching the profile's rule
+/// ordering). Policy paths are tilde-expanded exactly as the profile builder
+/// expands them so the comparison sees the same absolute paths the profile
+/// grants. Matching is component-wise, so `/data` never matches `/database`.
+fn is_working_directory_allowed(request: &ExecutionRequest, dir: &str) -> bool {
```
Comment (MGudgin) [inline src/backends/seatbelt/common/src/seatbelt_runner.rs:756]: ### Relative working directories are incorrectly replaced by the fallback

`process.cwd` currently accepts relative paths, which `Command::current_dir` resolves against the executor's host cwd. This check instead compares the unresolved relative string directly with absolute policy paths, so it always considers the path unreadable.

For example:

```json
{
  "process": {
    "cwd": "src"
  },
  "filesystem": {
    "readwritePaths": ["/repo"]
  }
}
```

If the executor is running from `/repo`, the intended child cwd is `/repo/src`. The new check compares `src` with `/repo`, fails, and launches the child from the fallback `/repo`. Commands using cwd-relative files then run from the wrong directory.

Similarly, `cwd: "."` can be replaced by the first read-write path even when the inherited cwd is already inside an allowed read-only path.

Could we resolve relative cwd values against `std::env::current_dir()` before normalization and containment checking? Please also add regression tests for `cwd: "."` and `cwd: "src"` when their resolved absolute paths are policy-readable.

File: src/backends/seatbelt/common/src/seatbelt_runner.rs [hunk 2]
```diff
@@ -708,6 +734,103 @@ fn resolve_working_directory(request: &ExecutionRequest) -> String {
...
+    // Expand `~` exactly as the profile builder does, then lexically fold `.` /
+    // `..` so the containment test compares the path the kernel actually
```
Comment (MGudgin) [inline src/backends/seatbelt/common/src/seatbelt_runner.rs:758]: ### The cwd readability check does not match the generated Seatbelt profile

`is_working_directory_allowed` only considers `readwritePaths` and `readonlyPaths`, but the generated Seatbelt profile also grants read access to several system locations unconditionally, including:

- `/bin`
- `/usr/bin`
- `/usr/lib`
- `/usr/share`
- `/System`
- `/Library`
- selected paths under `/private`

This means an explicit cwd such as `/usr/share` or `/Library/Application Support` may be readable under the actual profile, but this check still rejects it and silently launches from the first configured read-write or read-only path instead.

That changes command behavior unnecessarily—for example, `pwd`, relative arguments, and tools that discover configuration from the cwd will see a different directory even though the requested directory was already accessible.

Could we derive this decision from the same baseline path data used by `profile_builder` rather than maintaining a narrower second definition of readability? A regression test using a baseline-readable cwd such as `/usr/share` would also help prevent the two implementations from drifting.

To be fair, filesystem policy being deny-by-default means that we really need a way to 'include' the directories that are accessible by default on a given container platform, so that a) policy authors don't have to list them explicitly and b) the policy is still accurate. The policy 'spec' does allude to this, but we've not implemented it yet.

File: src/backends/seatbelt/common/src/seatbelt_runner.rs [hunk 3]
```diff
@@ -708,6 +734,103 @@ fn resolve_working_directory(request: &ExecutionRequest) -> String {
...
+    // resolves. Without this, `/work/../private` would spuriously match an
```
Comment (MGudgin) [inline src/backends/seatbelt/common/src/seatbelt_runner.rs:759]: ### Escape the requested cwd before writing it to the diagnostic log

The fallback message interpolates `request.working_directory` directly into a line-oriented log entry. Unix paths can contain control characters, including newlines, so a value such as:

```text
/outside-policy
ERROR: sandbox initialization failed
```

would create what appears to be a separate, forged log entry. This does not cross the sandbox boundary because the cwd comes from the caller's configuration, but it can make diagnostics misleading or difficult to parse reliably.

Could we log the path using escaped/debug formatting, such as `{:?}` or `escape_default()`, so each fallback remains a single well-formed log line?

File: src/backends/seatbelt/common/src/seatbelt_runner.rs [hunk 4]
```diff
@@ -708,6 +734,103 @@ fn resolve_working_directory(request: &ExecutionRequest) -> String {
...
+    // allowed `/work` even though it resolves outside it.
```
Comment (MGudgin) [inline src/backends/seatbelt/common/src/seatbelt_runner.rs:760]: ### Add coverage for the new fallback diagnostic

The new fallback behavior includes an operator-facing log message explaining why the requested cwd was replaced, but the tests only verify the returned directory. They do not verify that the diagnostic was emitted or that it identifies the requested and fallback paths.

Since these tests already use a buffered `Logger`, could we add assertions that:

- A rejected cwd logs both the requested directory and the selected fallback.
- An allowed cwd does not emit a fallback diagnostic.
- The logged paths are escaped so control characters cannot create additional log lines.

This would protect the primary diagnostic users have when a command unexpectedly starts in a different directory.

**Review by MGudgin — CHANGES_REQUESTED** [review]

**Comment by richiemsft** [comment]
@theelliotm is looking at it.
